cnf-tests/testsuites/e2esuite/multinetworkpolicy/multinetworkpolicy_sriov.go

```diff
@@ -0,0 +1,819 @@
+			eventually30s(nsX_podB).Should(Reach(nsX_podA, OnPort(port6666), ViaUDP))
+
+			// Not allowed
+			eventually30s(nsX_podB).ShouldNot(Reach(nsX_podA, OnPort(port6666), ViaTCP))
+			eventually30s(nsX_podB).ShouldNot(Reach(nsX_podA, OnPort(port5555), ViaUDP))
+		})
+
+		It("Allow access only to a specific UDP port from any pod", func() {
+
+			MakeMultiNetworkPolicy(TEST_NETWORK,
+				WithPodSelector(metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"pod": "a",
+					},
+				}),
+				WithIngressRule(multinetpolicyv1.MultiNetworkPolicyIngressRule{
+					Ports: []multinetpolicyv1.MultiNetworkPolicyPort{{
+						Port:     &port6666,
+						Protocol: &protoUDP,
+					}},
+				}),
+				CreateInNamespace(nsX),
+			)
+
+			// Allowed
+			eventually30s(nsX_podB).Should(Reach(nsX_podA, OnPort(port6666), ViaUDP))
+
+			// Not allowed
+			eventually30s(nsX_podB).ShouldNot(Reach(nsX_podA, OnPort(port5555), ViaTCP))
+			eventually30s(nsX_podB).ShouldNot(Reach(nsX_podA, OnPort(port5555), ViaUDP))
+			eventually30s(nsX_podB).ShouldNot(Reach(nsX_podA, OnPort(port6666), ViaTCP))
+		})
+	})
+
+	Context("IPv6", func() {
+		It("ALLOW traffic to nsX_podA only from nsX_podB", func() {
+
+			Skip("IPv6 not yet supported by multinetworkpolicy iptables - https://github.com/k8snetworkplumbingwg/multi-networkpolicy-iptables/issues/22")
+
+			MakeMultiNetworkPolicy(TEST_NETWORK,
+				WithPodSelector(metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"pod": "a",
+					},
+				}),
+				WithIngressRule(multinetpolicyv1.MultiNetworkPolicyIngressRule{
+					From: []multinetpolicyv1.MultiNetworkPolicyPeer{{
+						PodSelector: &metav1.LabelSelector{
+							MatchLabels: map[string]string{
+								"pod": "b",
+							},
+						},
+					}},
+				}),
+				CreateInNamespace(nsX),
+			)
+
+			// IPv4
+			eventually30s(nsX_podB).Should(Reach(nsX_podA, ViaIPv4))
+			eventually30s(nsX_podC).ShouldNot(Reach(nsX_podA, ViaIPv4))
+			eventually30s(nsX_podB).Should(Reach(nsX_podC, ViaIPv4))
+
+			// IPv6
+			eventually30s(nsX_podB).Should(Reach(nsX_podA, ViaIPv6))
+			eventually30s(nsX_podC).ShouldNot(Reach(nsX_podA, ViaIPv6))
+			eventually30s(nsX_podB).Should(Reach(nsX_podC, ViaIPv6))
+		})
+	})
+
+})
+
+func cleanPodsInNamespaces(argNamespaces ...string) {
+	for _, ns := range argNamespaces {
+		namespaces.CleanPods(ns, client.Client)
+	}
+}
+
+func cleanMultiNetworkPoliciesFromNamespace(namespace string) {
+	err := client.Client.MultiNetworkPolicies(namespace).
+		DeleteCollection(context.Background(), metav1.DeleteOptions{}, metav1.ListOptions{})
+	Expect(err).ToNot(HaveOccurred())
+
+	Eventually(func() int {
+		ret, err := client.Client.MultiNetworkPolicies(namespace).
+			List(context.Background(), metav1.ListOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		return len(ret.Items)
+	}, 30*time.Second, 1*time.Second).Should(BeZero())
+}
+
+func createNamespaces(argNamespaces ...string) {
```

Comment:
please move this function to namespace

Reply:
Removed and exploded, it wouldn't gain much readability if I moved to namespace package.